File: VB/Scaffolding.ValidationErrors/DepartmentContextDataModel/Model/IDepartmentContextUnitOfWork.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.ValidationErrors.Common.Utils
Imports Scaffolding.ValidationErrors.Common.DataModel
Imports Scaffolding.ValidationErrors.Common.DataModel.EntityFramework
Imports Scaffolding.ValidationErrors.Model

Namespace Scaffolding.ValidationErrors.DepartmentContextDataModel
	Public Interface IDepartmentContextUnitOfWork
	Inherits IUnitOfWork
		Function HasChanges() As Boolean
		ReadOnly Property Departments() As IDepartmentRepository
	End Interface
End Namespace

File: CS/Scaffolding.ValidationErrors/DepartmentContextDataModel/Model/IDepartmentContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.ValidationErrors.Common.Utils;
using Scaffolding.ValidationErrors.Common.DataModel;
using Scaffolding.ValidationErrors.Common.DataModel.EntityFramework;
using Scaffolding.ValidationErrors.Model;

namespace Scaffolding.ValidationErrors.DepartmentContextDataModel {
    public interface IDepartmentContextUnitOfWork : IUnitOfWork {
        bool HasChanges();
        IDepartmentRepository Departments { get; }
    }
}
